python/ray/autoscaler/_private/vsphere/config.py

```diff
@@ -193,34 +196,23 @@ def update_vsphere_configs(config):
     #               resources
     # This enables us to access the field during node creation.
     # The same happens for head node too.
-    worker_node_config["resources"] = worker_node["resources"]
     head_node_config["resources"] = head_node["resources"]
-
     head_resource_pool = None
     if "resource_pool" in head_node_config:
         head_resource_pool = head_node_config["resource_pool"]
 
-    # by default create worker nodes in the head node's resource pool
-    worker_resource_pool = head_resource_pool
-
-    # If different resource pool is provided for worker nodes, use it
-    if "resource_pool" in worker_node_config:
-        worker_resource_pool = worker_node_config["resource_pool"]
-
-    worker_node_config["resource_pool"] = worker_resource_pool
-
-    worker_datastore = None
-
-    if "datastore" in head_node_config and head_node_config["datastore"]:
-        worker_datastore = head_node_config["datastore"]
+    for worker_node in worker_nodes:
+        worker_node["node_config"]["resources"] = worker_node["resources"]
 
-    if "datastore" in worker_node_config and worker_node_config["datastore"]:
-        worker_datastore = worker_node_config["datastore"]
+    for worker_node_config in worker_node_configs:
+        # if the resource pool for the worker node is unspecified then let it be the
+        # head node's resource pool
 
-    worker_node_config["datastore"] = worker_datastore
```

Comment:
We don't need `datastore` anymore?

Reply:
Please forgive the original code which is hard to read.

Before the change, the logic is: if the datastore of the worker node is unset, use the head node's datastore by default.

But, what we actually want is: if the datastore of the worker node is unset, use the frozen VM's datastore.

So I make this change. We left the datastore empty for the worker node. It will automatically use the datastore of the frozen VM with this empty parameter.